src/DetailsView/components/details-view-command-bar.tsx

```diff
@@ -229,11 +242,49 @@ export class DetailsViewCommandBar extends React.Component<
 
     private renderLoadAssessmentButton = (): JSX.Element | null => {
         if (this.props.featureFlagStoreData.saveAndLoadAssessment) {
-            return <LoadAssessmentButton {...this.props} />;
+            return (
+                <LoadAssessmentButton
+                    {...this.props}
+                    handleLoadAssessmentButtonClick={this.handleLoadAssessmentButtonClick}
+                    onClose={this.toggleLoadAssessmentDialog}
+                />
+            );
         }
         return null;
     };
 
+    private renderLoadAssessmentDialog = (): JSX.Element => {
+        return this.props.switcherNavConfiguration.loadAssessmentDialogFactory({
+            ...this.props,
+            isOpen: this.state.isLoadAssessmentDialogOpen,
+            prevTab: this.props.assessmentStoreData.persistedTabInfo,
+            loadedAssessmentData: this.state.loadedAssessmentData,
+            tabId: this.props.tabStoreData.id,
+            onClose: this.toggleLoadAssessmentDialog,
+        });
+    };
+
+    private toggleLoadAssessmentDialog = () => {
```

Comment:
This and the two methods below do not have tests validating them.

Reply:
I've added tests to validate these.